packages/swimlane/data_stream/swimlane_api/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,197 @@
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.Path
+        target_field: url.path
+        ignore_missing: true
+    - grok:
+        field: swimlane.audit_log.SourceIp
+        patterns:
+          - "::ffff:%{IPV4:source.ip}"
+          - "%{IPV4:source.ip}"
+          - "%{IPV6:source.ip}" 
+        ignore_failure: true
+    - geoip:
+        field: source.ip
+        target_field: source.geo
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.Tenant
+        target_field: cloud.origin.tenant.id
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.LogCategory
+        target_field: log.category
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.LogFeatureCategory
+        target_field: log.feature_category
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.LogSource
+        target_field: log.source
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.LogType
+        target_field: log.type
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.Description
+        target_field: message
+        ignore_missing: true
+    - set:
+        field: message
+        value: "{{message}}: {{swimlane.audit_log.NewValue.Name}} with Id: {{swimlane.audit_log.NewValue.Id}}"
+        if: 'ctx.swimlane.audit_log.NewValue != null && ctx.swimlane.audit_log.NewValue.Name != null && ctx.swimlane.audit_log.NewValue.Id != null'
+    - rename:
+        field: swimlane.audit_log.UserId
+        target_field: user.id
+        ignore_missing: true
+    - script:
+        lang: painless
+        tag: Add ECS categorization
+        params:
+          Create:
+            type:
+              - creation
+          Read:
+            type:
+              - access
+          Update:
+            type:
+              - change
+          Delete:
+            type:
+              - deletion
+          Login:
+            type:
+              - start
+          Logout:
+            type:
+              - end   
+        source: >-
+          ctx.event.kind = 'event';
+          ctx.event.type = ['info'];
+          if (ctx?.swimlane?.audit_log?.ActionType == null) {
+              return;
+          }
+          if (params.get(ctx?.swimlane?.audit_log?.ActionType) == null) {
+              return;
+          }
+          def hm = new HashMap(params.get(ctx?.swimlane?.audit_log?.ActionType));
+          hm.forEach((k, v) -> ctx.event[k] = v);
+    - rename:
+        field: swimlane.audit_log.Category
+        target_field: log.category
+        ignore_missing: true
+    - rename:
+        field: swimlane.audit_log.Endpoint
+        target_field: url.path
+        ignore_missing: true
+    # drop all events that are related to the hubs/record endpoint as it's too noisy
+    - drop:
+        if: 'ctx?.url?.path != null && ctx.url?.path.toLowerCase().contains("hubs/record")'
```

```suggestion
        if: ctx.url?.path != null && ctx.url.path.toLowerCase().contains("hubs/record")
```
(`ctx` always exists, so we don't need the null-safe there, and in order to reach the second clause of the conjunction, `ctx.url` must be non-null due to Java bool expression short circuiting, similar below)